SQL.js

```diff
@@ -36,14 +40,30 @@ class SqlStatement {
 
   get text () {
     let text = this.strings[0]
+    let parameter = 0
 
     for (let i = 1; i < this.strings.length; i++) {
-      text += '$' + i + this.strings[i]
+      const value = this._values[i - 1]
+
+      if (isRawValue(value)) {
+        text += value.__raw + this.strings[i]
+      } else {
+        parameter++
+        text += '$' + parameter + this.strings[i]
+      }
     }
 
     return text.replace(/^\s+|\s+$/mg, '')
   }
 
+  get values () {
+    return this._values.filter(v => !isRawValue(v))
+  }
+
+  get rawValues () {
+    return this._values.filter(isRawValue)
+  }
```

Comment:
doing these live are going to be expensive. I would prefer if we didn't store this at all and did the generation/computation ahead of time rather than on-the-fly.